src/components/domain/ProductCard/index.tsx

```diff
@@ -26,6 +23,14 @@ const ProductCard = ({ product }: Props) => {
     );
   }
 
+  const { sendRequest, loading = false } = useHttp<Cart>(cartApi.postAddCart);
+
+  const handleClickCart = async () => {
+    const newCart = await sendRequest(product);
+    if (!newCart) return;
+    onOpenModal?.();
```

Comment:
`async & await` 를 사용 중이라 반환에 대한 부분이 비동기에서 어떻게 처리될지도 체크해보는게 좋겠어요!

```suggestion
    if (newCart) {
      onOpenModal?.();
    }
```